__tests__/ExpensiMark-Markdown-Truncate-test.js

```diff
@@ -0,0 +1,109 @@
+        const result = parser.truncateHTML(html, limit, { ellipsis: false });
+        expect(result).toBe('This is a <strong>long</strong> text that exceeds the character limit. It contains multiple sente');
+    });
+
+    test('should not truncate HTML in the middle of a word', () => {
+        const markdown = 'This is a *long* text that exceeds the character limit. It contains a very long word at the end: antidisestablishmentarianism.';
+        const limit = 90;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, { ellipsis: true });
+        expect(result).toBe('This is a <strong>long</strong> text that exceeds the character limit. It contains a very long word at the ...');
+    });
+
+    test('should handle HTML with multiple markdown elements', () => {
+        const markdown = 'This is a *long* text with _multiple_ markdown ~elements~. It includes *bold*, _italic_, and ~strikethrough~ formatting. The truncation should preserve the markdown syntax.';
+        const limit = 80;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, { ellipsis: true });
+        expect(result).toBe('This is a <strong>long</strong> text with <em>multiple</em> markdown <del>elements</del>. It includes <strong>bold</strong>, <em>italic</em>, a...');
+    });
+
+    test('should handle HTML with nested markdown elements', () => {
+        const markdown = 'This is a *long _nested_ markdown* text. It contains *_bold italic_* and ~_strikethrough italic_~ formatting. The truncation should handle the nesting correctly.';
+        const limit = 80;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, { ellipsis: true });
+        expect(result).toBe('This is a <strong>long <em>nested</em> markdown</strong> text. It contains <strong><em>bold italic</em></strong> and <del><em>strikethrough i...</em></del>');
+    });
+
+    test('should handle HTML with links', () => {
+        const markdown = 'This is a text with [a link](https://example.com) that exceeds the limit. The link should be preserved in the truncated text. Additionally, it includes [another link](https://example.org) for testing purposes.';
+        const limit = 80;
+        const html = parser.replace(markdown);
+        const result = parser.truncateHTML(html, limit, { ellipsis: true });
+        expect(result).toBe('This is a text with <a href=\"https://example.com\" target=\"_blank\" rel=\"noreferrer noopener\">a link</a> that exceeds the limit. The link should be preserved ...');
+    });
+
+    test('should handle HTML with code blocks', () => {
+        const html = 'This is a text with <blockquote>code block</blockquote> that exceeds the limit. The code block should be preserved in the truncated text. Here is an example code block:\n\n```javascript\nconsole.log("Hello, world!");\n```\n\nThe truncation should handle code blocks correctly.';
```

Comment:
Not sure of the purpose of the `<blockquote>` tag as part of this test and the current text is long enough that the code block is not preserved. 
```suggestion
        const html = 'This is a text with code block that exceeds the limit. The code block should be preserved in the truncated text. Here is an example code block:\n\n```javascript\nconsole.log("Hello, world!");\n```\n\nThe truncation should handle code blocks correctly.';
```